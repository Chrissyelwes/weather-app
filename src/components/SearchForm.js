import React from "react";
import PropTypes from "prop-types";

function SearchForm({ searchText, setSearchText, onSubmit }) {
  const handleInputChange = (event) => setSearchText(event.target.value);

  return (
    <form className="search-form">
      <input
        className="input-field"
        type="text"
        placeholder="Enter city..."
        onChange={handleInputChange}
        value={searchText}
      />
      <button
        className="search-button"
        type="submit"
        data-testid="button-text"
        onClick={onSubmit}
      >
        Search
      </button>
    </form>
  );
}

export default SearchForm;

SearchForm.propTypes = {
  searchText: PropTypes.string.isRequired,
  setSearchText: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};
