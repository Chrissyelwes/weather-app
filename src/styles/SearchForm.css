.input-field {
    padding: 5px 28px;
    border: none;
    font-family: 'Roboto', sans-serif;
    color: #7E5444;
    margin-bottom: .625rem;
}

/* .search_button {
    padding: 5px 28px;
    font-family: 'Raleway', sans-serif;
    color: whitesmoke;
    text-align: center;
    border: none;
    transition-duration: 0.4s;
}

.search_button:hover {
    background-color: whitesmoke;
    color: whitesmoke;
} */

.search_button {
    transition: all .5s ease;
    padding: 5px 28px;
    font-family: 'Raleway', sans-serif;
    background-color: #A3904C;
    color: whitesmoke;
    text-align: center;

    border: none;
    transition-duration: 0.4s;
}

.search_button:hover {
    background-color: #7E5444;
}

@media only screen and (max-width: 375px) {
    .input-field {
        padding: 5px 28px;
        border: none;
        font-family: 'Roboto', sans-serif;
        color: #7E5444;
        margin-bottom: .625rem;
        width: 150px;
    }

    .search_button {
        font-size: medium;
    }
}

