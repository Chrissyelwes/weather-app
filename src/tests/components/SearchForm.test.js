import React from "react";
import { render, screen } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

describe("Searchform", () => {
  const validProps = {
    searchText: "",
    setSearchText: () => {},
    onSubmit: () => {},
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSubmit}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("displays the correct text on button", () => {
    render(<SearchForm />);
    const searchButton = screen.getByRole("button", { name: /search/i });

    expect(searchButton).toHaveTextContent("Search");
  });

  it("has a class", () => {
    render(<SearchForm />);
    const searchButton = screen.getByRole("button");

    expect(searchButton).toHaveClass("search_button");
  });
});
