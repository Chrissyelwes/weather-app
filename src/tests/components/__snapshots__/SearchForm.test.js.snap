// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchform renders correctly 1`] = `
<DocumentFragment>
  <form
    class="search-form"
  >
    <input
      class="input-field"
      placeholder="Enter city..."
      type="text"
      value=""
    />
    <button
      class="search_button"
      data-testid="button-text"
      type="submit"
    >
      Search
    </button>
  </form>
</DocumentFragment>
`;
